test/unit/Clustering/GMUSimpleClusterAlgorithmTest.swift

```diff
@@ -0,0 +1,59 @@
+/* Copyright (c) 2020 Google Inc.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+import XCTest
+@testable import GoogleMapsUtils
+
+class GMUSimpleClusterAlgorithmTest: GMUClusterAlgorithmTest {
+  
+  private var clustersCount: Int!
+  private var zoom: Float!
+  
+  override func setUp() {
+    clustersCount = 10
+    zoom = 3
+    super.setUp()
+  }
+  
+  override func tearDown() {
+    clustersCount = nil
+    zoom = nil
+    super.tearDown()
+  }
+  
+  func testClustersAtZoomWithDefaultClusterCount() {
+    let simpleClusterAlgorithm = GMUSimpleClusterAlgorithm()
+    simpleClusterAlgorithm.add(self.simpleClusterItems())
+    simpleClusterAlgorithm.add(self.simpleClusterItems())
+    simpleClusterAlgorithm.add(self.simpleClusterItems())
+    let clusterItems = simpleClusterAlgorithm.clusters(atZoom: zoom)
+    XCTAssertEqual(clustersCount, clusterItems.count)
```

Comment:
It's not obvious to me that this should equal 10 when there were 12 items that were added. Can you give a bit more context?

Reply:
When I was going through the code, here are my findings:

No matter how many items it's added > 10 like 12-20, the method will always return `kClusterCount` which is equal to 10[declared here `static const NSUInteger kClusterCount = 10;`]

If it's lesser than 10, then it will be like 4-8 it will give the actual input, like if input = 4, the output is also 4. but for >10 the if input > 10[like 12-20], output = 10.

I hope you got my thoughts. Please clarify if I misunderstood also!